File: Backend/Models/Project.cs
using System.ComponentModel.DataAnnotations;

namespace PortfolioWebsiteBackend.Models
{
    public class Project
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string? ImageUrl { get; set; }

        public string? ProjectLink { get; set; }

        public string? GithubLink { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Backend/src/models/project.ts
export interface Project {
  id: number;
  title: string;
  description: string | null;
  imageUrl: string | null;
  projectLink: string | null;
  githubLink: string | null;
  createdAt: string;
}

export interface CreateProjectDto {
  title: string;
  description?: string | null;
  imageUrl?: string | null;
  projectLink?: string | null;
  githubLink?: string | null;
}

export interface UpdateProjectDto {
  title: string;
  description?: string | null;
  imageUrl?: string | null;
  projectLink?: string | null;
  githubLink?: string | null;
}
